ecl/hqlcpp/hqlcpputil.cpp

```diff
@@ -442,3 +442,14 @@ bool worthGeneratingRowAsSingleActivity(IHqlExpression * expr)
         expr = expr->queryChild(0);
     }
 }
+
+bool isNonConstantAndQueryInvariant(IHqlExpression * expr)
+{
+    if (expr->isConstant())
```

Comment:
can this be true, and hasDynamicFilename(info.queryFile()) also be true?

Reply:
@jakesmith yes I think so.  DYNAMIC(filename) means resolve the filename each time - even if it is constant.  Is that ok?

Reply:
@ghalliday, yes I think so, I will only assume invariant as this logic implies, when the FFinvariantfilename is present.

Reply:
Seems like a weird condition to want to know. Why are constant filenames not considered context-independent?

Reply:
@richardkchapman  Not sure.  I think it was purely to minimize the number of changes in the helper files - since almost all examples are constant (and simplify my comparison when regression checking).  I think we could equally include constant expressions - I can change if you prefer.